great_expectations/expectations/expectation.py

```diff
@@ -318,12 +318,12 @@ class Expectation(pydantic.BaseModel, metaclass=MetaExpectation):
     class Config:
         arbitrary_types_allowed = True
         smart_union = True
-        extra = pydantic.Extra.allow
+        extra = pydantic.Extra.forbid
 
     id: Union[str, None] = None
     meta: Union[dict, None] = None
     notes: Union[str, None] = None
-    result_format: Union[ResultFormat, ResultFormatDict] = ResultFormat.BASIC
+    result_format: Union[ResultFormat, dict] = ResultFormat.BASIC
```

Comment:
We're deleting `result_format` support on Expectations moving forward - no point in strongly typing this if its gonna go in a follow-up ticket